pyproject.toml

```diff
@@ -12,7 +12,8 @@ dependencies = [
     "google-cloud-alloydb-connector[asyncpg]>=1.2.0, <2.0.0",
     "google-cloud-storage>=3.0.0, <3.1.0",
     "langchain-core>=0.2.36, <1.0.0",
-    "numpy>=1.24.4, <2.0.0",
+    "numpy>=2.2.2, <2.3.0; python_version > '3.9'",
```

Comment:
3.9 can also still use the bottom of the range 1.24.4. It's not forced to start at 2.2.2